packages/core/content-type-builder/server/controllers/validation/content-type.js

```diff
@@ -122,8 +125,10 @@ const forbiddenContentTypeNameValidator = () => {
   };
 };
 
-const alreadyUsedContentTypeName = (isEdition) => {
-  const usedNames = _.flatMap(strapi.contentTypes, (ct) => [ct.singularName, ct.pluralName]);
```

Comment:
The bug mentioned in my comment on the tests file. We already should have been comparing singularName and pluralName, but instead we were getting an array of "undefined".